### DISCUSSION TITLE
How can I use IME mode(only English) when I want?

### DISCUSSION THREAD

**Discussion by GGongM** [cefsharp/CefSharp#4490] [Q&A] [answered]
Hi,
I'm trying to cefshrap with winform.
I want to receive input in English only when I click a specific text box on the web.

So I use imm32.dll

 [DllImport("imm32.dll")]
 internal static extern bool ImmReleaseContext(IntPtr hWnd, IntPtr hIMC);

 [DllImport("imm32.dll")]
 public static extern Boolean ImmSetConversionStatus(IntPtr hIMC, Int32 fdwConversion, Int32 fdwSentence);

private static ChromiumWebBrowser m_browser;
public static bool ChangeIME(bool toogle) {
         IntPtr hWnd = m_browser.Handle; 
         IntPtr himc = ImmGetContext(hWnd);
         if (!toggle) {   
                 ret = ImmSetConversionStatus(himc, 0, 0);  //IME - English
         }
         else {
                ret = ImmSetConversionStatus(himc, 1, 0);  //IME -  Korean
         }
         ImmReleaseContext(hWnd, himc)
}

Get the handle of the ChromiumWebBrowser well and the return value after conversion is true, but it does not change.
So I tried with the handle of the main form that floated the web, but it's the same.

How can I get English input by changing IME mode like pressing 'Language toggle key' ?

**Comment by amaitland** [comment, marked as answer]
> I'm trying to cefshrap with winform.

`Chromium` handles `IME`. I'm not sure what you are attempting is possible.

Try asking on https://www.magpcss.org/ceforum/index.php

> IntPtr hWnd = m_browser.Handle;

That's the `WinForms` hande, the `CEF Browser` handle is obtained by a different method, see https://github.com/cefsharp/CefSharp/blob/4699eb5e0e6ba71c4f06442c5abfce5e20012c72/CefSharp.WinForms/WebBrowserExtensions.cs#LL57C62-L57C62
